svm/api.go

```diff
@@ -134,3 +138,67 @@ func (svm *SVM) ValidateAndAddTxToPool(tx *types.Transaction) error {
 	}
 	return nil
 }
+
+// HandleGossipTransaction wraps around HandleTransaction,
+// which handles data received on the transactions gossip channel.
+func (svm *SVM) HandleGossipTransaction(ctx context.Context, _ p2p.Peer, msg []byte) pubsub.ValidationResult {
+	tx, err := types.BytesToTransaction(msg)
+	if err != nil {
+		svm.state.With().Error("SVM couldn't parse incoming transaction", log.Err(err))
+		return pubsub.ValidationIgnore
+	}
+
+	if err = svm.handleTransaction(tx, true); err != nil {
+		svm.state.With().Error("handle transaction", log.Err(err))
+		return pubsub.ValidationIgnore
+	}
+
+	return pubsub.ValidationAccept
+}
+
+// HandleTxSyncData handles data received on transaction sync.
+func (svm *SVM) HandleTxSyncData(data []byte) error {
```

Comment:
Let's switch to more consistent naming: since we have `HandleGossipTransaction`, let's call this `HandleSyncTransaction`.